test/tests/functional/pbs_holidays.py

```diff
@@ -146,3 +170,117 @@ def test_only_year(self):
+
+        self.check_prime()
+
+        # Verify that scheduler logged a message for out of date file
+        msg = "holidays;The holiday file is out of date; please update it."
+        self.scheduler.log_match(msg, starttime=ctime)
+
+    def test_commented_holidays_file(self):
+        """
+        Test that the scheduler assumes all prime-time if the holidays
+        file is completely commented out
+        """
+        ctime = int(time.time())
+        self.scheduler.holidays_delete_entry('a')
+
+        content = """* YEAR 1970
+*  weekday 0600  1730
+*  saturday  none  all
+*  sunday  none  all"""
+
+        self.scheduler.holidays_write_file(content=content)
+
+        # Verify that the holidays file has 4 lines
+        h_content = self.du.cat(filename=self.scheduler.holidays_file,
+                                sudo=True)['out']
+        self.assertEqual(len(h_content), 4)
+
+        self.check_prime()
+
+        # Verify that scheduler didn't log a message for out of date file
+        msg = "holidays;The holiday file is out of date; please update it."
+        self.scheduler.log_match(msg, starttime=ctime, existence=False,
+                                 max_attempts=5)
+
+    def test_non_prime(self):
+        """
+        Test that non-prime time set via holidays file works correctly
+        """
+        self.scheduler.holidays_delete_entry('a')
+
+        self.scheduler.holidays_set_year(self.cur_year)
+
+        # Set all days of the week as prime time except today
+        today_idx = datetime.datetime.today().weekday()
+        today = self.days[today_idx]
+        for i in range(7):
+            if i != today_idx:
+                self.scheduler.holidays_set_day(self.days[i], "all", "none")
+
+        # Set today as all non-prime time
+        self.scheduler.holidays_set_day(today, "none", "all")
+        self.check_nonprime()
+
+    def test_missing_year(self):
+        """
+        Test that scheduler assumes all prime time if the year entry is
+        missing from holidays file
+        """
+        ctime = int(time.time())
+        self.scheduler.holidays_delete_entry('a')
+
+        # Create a holidays file with no year entry and all days set to
+        # 24hrs non-prime time
+        content = """*  weekday none  all
+*  saturday  none  all
+*  sunday  none  all"""
+
+        self.scheduler.holidays_write_file(content=content)
+
+        # Verify that the holidays file has 3 lines
+        h_content = self.du.cat(filename=self.scheduler.holidays_file,
+                                sudo=True)['out']
+        self.assertEqual(len(h_content), 3)
+
+        self.check_prime()
+
+        # Verify that scheduler didn't log a message for out of date file
+        msg = "holidays;The holiday file is out of date; please update it."
+        self.scheduler.log_match(msg, starttime=ctime, existence=False,
+                                 max_attempts=5)
```

Comment:
One of the use cases of commenting out the holiday file is so the scheduler doesn't add prime status events to the calendar.  Consider writing a test to check for that.  To do it, you need to set log_filter to 2048.  You then add a job that takes up all the resources without requesting walltime.  You then add a second job that will get calendared.  You look in the log that no policy change events happened.